tests/RedisTest.php

```diff
@@ -5288,6 +5288,46 @@ public function testMultipleConnect() {
         }
     }
 
+    public  function testSession_regenerateSessionId_noLock_noDestroy() {
+        $this->setSessionHandler();
+        $sessionId = $this->generateSessionId();
+        $writeSuccessful = $this->startSessionProcess($sessionId, 0, false, 300, true, null, -1, 1, 'bar');
+
+        $newSessionId = $this->regenerateSessionId($sessionId);
+	
+        $this->assertEquals('bar', $this->getSessionData($newSessionId));
```

Comment:
Maybe it makes sense to add additional assertions?
* Old data is deleted/destroyed
* New session ID differs from old one